Update catver.ini with more complete information

Wilstorm opened the pull request:
Update catver.ini with more complete information and new entries. 

See issue https://github.com/libretro/mame2003-plus-libretro/issues/759 for more information.

Thank you wanting to make a contribution to this project!

Please note that by contributing code or other intellectual to this project you are allowing the project to make unlimited use of your contribution. As with the rest of the project, new contributions will be made available freely under the classic MAME Non-Commercial License.

**This license can be viewed at https://raw.githubusercontent.com/libretro/mame2003-plus-libretro/master/LICENSE.md**.

markwkidd commented:
This is a major undertaking, which I say from first person knowledge having contemplated doing this myself and deciding not to because it was too much work. "They said it couldn't be done!" Congrats!

That being said, there is one concern that I can see so far: if I recall correctly, my updates to the catver.ini have maintained the standard of the MAME 0.78 era in the sense that non-working games are labeled "Uncategorized"

While adding genres to the uncategorized games might be more accurate, some filtering software and scripts use the "uncategorized" label to help filter these games, so adopting the more modern approach of listing genres even for these games might have unintended consequences.

I'm writing this comment to create a little space to make sure it's the best thing to flip the Uncategorized listings over to indicating genre.

Wilstorm commented:
@markwkidd - Thanks, I didn't do the work on catver.ini the credit for all the work goes to @Redemp. You make a good point, maybe this is a bit premature and should be closed for now?

markwkidd commented:
Nah, I hope we don't need to close the PR. This is very much in the spirit
of my own contributions, and I would like to see it merged. I also do
remember making some changes in the past that had unintended consequences
which is all I'm trying to think about LOL

On Mon, Jul 6, 2020 at 6:01 PM Wilstorm <notifications@github.com> wrote:

> @markwkidd <https://github.com/markwkidd> - Thanks, I didn't do the work
> on catver.ini the credit for all the work goes to @Redemp
> <https://github.com/Redemp>. You make a good point, maybe this is a bit
> premature and should be closed for now?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/libretro/mame2003-plus-libretro/pull/829#issuecomment-654487270>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEVGC5R663B3IQFA3CDC4I3R2JCSBANCNFSM4OR7RD6A>
> .
>


-- 
Mark W. Kidd
http://facebook.com/markwkidd
(606)536-0115

Wilstorm commented:
@markwkidd - Ok, I appreciate your insight as you have a much better idea than I do on if this is a good update for mame2003-plus.

Redemp commented:
Couldn't this information be extracted from src/drivers/ ?

Example

```
src/drivers/zn.c

GAMEX( 1995, ts2,      cpzn1,    znqsound, zn, zn,   ROT0, "Capcom/Takara", "Battle Arena Toshinden 2 (USA 951124)", GAME_UNEMULATED_PROTECTION | GAME_NOT_WORKING )

```

```
metadata/mame2003-plus.xml

	<game name="ts2" romof="cpzn1">
		<description>Battle Arena Toshinden 2 (USA 951124)</description>
```

markwkidd commented:
@Redemp (I didn't realize you also had the name Rion. Hi! Thanks again for this!) you are correct that the same information is in the driver information. In fact, one of my projects a year or two ago was to make sure that the catver.ini was synchronized to the driver information in regards to not working games.

Let me say more about my concern. I'm concerned that there may be some frontend management scripts out there for emulationstation that rely on the old approach to not working games. There may be utilities that only look at the catver.ini but don't parse the DAT file where they could find out a game isn't working.

My attitude, especially as someone who isn't very involved right now, is to go ahead and merge this and wait to see if anyone has a problem. :) The more I write about this, the less it seems like a big deal, and maybe it's good to go ahead and merge.

Redemp commented:
I'm in the process of filtering out the non working games from the src/drivers/ and will then post a list with all the roms. 
This could then be added to the catver.ini as you see fit.

Wilstorm commented:
@Redemp - Sorry, I also hadn’t realized this Rion either. I didn’t mean to overstep by no means. I was just trying to help by updating the catver.ini with all the nice work you put into it. I saw Arcadez mention doing some updates to the metadata and thought what the heck. You want me to close this PR until you finish all the work you would like to do so it’s more complete?

Redemp commented:
No problem. Nobody is overstepping anyone. 
It's my own fault that i have two usernames.
Opensource is a collaboration from multiple sources as far as I'm concerned (as far as i know). I can't exactly say when i will be finished, but it should be to long. The question is more of how it should be structured? 

Should they still be marked as "Uncategorized" or for example 

ts2=Fighter / Versus * Not Working *

Wilstorm commented:
@Redemp @markwkidd - Ok, I just wanted to make sure! :)

I like the idea of "Uncategorized" for games that don't work for filtering makes sense. Not point in pulling in games that don't work.

Is it possible to add an additional "fake" category like "Uncategorized" that could be 'not' filtered. So I am looking for "Fighter" 'not' "Uncategorized". Basically you can filter on any genre but have a 'not' check box to filter the "Uncategorized" for any search. That way if the game changes status to working it would be easy enough to just remove the "Uncategorized" category from the catver.ini. You would also have the original correct category already in place.

```ts2=Fighter / Uncategorized```

On the other hand if the convention is Uncategorized as the only genre that could be an issue with existing uses. 

I'm just throwing ideas out there but honestly I'm ok with whatever you decide as I would utilize the new catver.ini. Mark you did a lot of the original work would you have a suggestion or idea on the best way to proceed?

Redemp commented:
So after searching thru 728 drivers the magical number was 421 games marked as GAME_NOT_WORKING.

I have now uploaded a new catver.ini with 421 games marked as **Uncategorized**

```
Before
1944j=Shooter / Flying Vertical
After
1944j=**Uncategorized**
```
You can find it with the files "GAME_NOT_WORKING_rom_names" (Raw list) and a backup file "GAME_NOT_WORKING_Category" of how it was before for reference if the game will ever be added as working. 

[Catver.ini NEW](https://drive.google.com/drive/folders/14ikB_inm6-2xFX0R-iu_A0wYbnNl5EpY?usp=sharing)

Wilstorm commented:
@Redemp - The file looks amazing and I see many corrections for uncategorized, etc. great job! It looks real good to me. 

The only difference is I see ```**Uncategorized**``` vs ```Uncategorized```. Which would be really easy to remove if the extra asterisks cause issues with filters. 

@markwkidd how does this new file look to you? Are the Astericks ok or better to remove?

markwkidd commented:
The MAME 0.78 styles the uncategorized tag as simply `Uncategorized`. It's
arbitrary but I do prefer to maintain that format in case there are any
sorting scripts or apps out there that have hardcoded that string.

On Wed, Jul 8, 2020 at 12:36 PM Wilstorm <notifications@github.com> wrote:

> @Redemp <https://github.com/Redemp> - The file looks amazing and I see
> many corrections for uncategorized, etc. great job! It looks real good to
> me.
>
> The only difference is I see **Uncategorized** vs Uncategorized. Which
> would be really easy to remove if the extra asterisks cause issues with
> filters.
>
> @markwkidd <https://github.com/markwkidd> how does this new file look to
> you? Are the Astericks ok or better to remove?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/libretro/mame2003-plus-libretro/pull/829#issuecomment-655627839>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEVGC5STAA4BCDAMYUK7FTTR2SN7RANCNFSM4OR7RD6A>
> .
>


-- 
Mark W. Kidd
http://facebook.com/markwkidd
(606)536-0115

Redemp commented:
@markwkidd

Ok.
```
1944j=**Uncategorized**
To
1944j=Uncategorized
```

I will fix it when i come home from work 😊

Wilstorm commented:
@markwkidd - Once that is complete you want to me submit a PR with the corrected file so you can take a look at the changes side by side?

Wilstorm commented:
@markwkidd - I have to admit I'm not very good at doing PR's and making mistakes. I could definitely use some help as I can't find the answers.

I added in an extra file change into the catver.ini PR (which shouldn't have been there) and then I reverted it but it shows 3 changes in the PR but I don't know how to get rid of the others unless you're not supposed to.

Also I have no idea how to update my fork. The last time I changed the base and head around but it came across as a pull request to the core here. Is it possible to update your fork without making it a pull request? That seems like a bother and should be able to sync forks.

Lastly is the best method (hierarchy?) to updating is create a fork, then a branch with the updates? Basically create a new branch for each update to keep the PR's separate?

If not someone with more experience can take it too if it's to much to explain as I've definitely spent the time trying to understand and I'm missing some things.



I have to admit I'm over my head here and shooting from the hip... =/

Redemp commented:
@Wilstorm @markwkidd @arcadez2003 @Darknior

File fixed. This should hopefully be the last one : )

[catver.ini](https://drive.google.com/drive/folders/14ikB_inm6-2xFX0R-iu_A0wYbnNl5EpY?usp=sharing)

Wilstorm commented:
@Redemp - It looks good! Thanks a bunch for the work put into this file! Really appreciative!

Wilstorm commented:
@markwkidd @Redemp - I think I am going to close this PR and let someone with a better knowledge of Github submit this update if you guys have no objections. I tried to learn it so I could participate as part of this Libretro project with some of the simpler updates of the metadata but I'm afraid I'm making more of mess than necessary. I spent a about 3 years watching @markwkidd @arcadez2003 and @grant2258 do some pretty amazing things and just wanted to return some of that effort. Anyway I'll leave it to the guru's here.
